specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp

```diff
@@ -209,3 +243,109 @@ union PublicNetworkAccess {
   /** If set, the private endpoints are the exclusive access method. */
   "Disabled",
 }
+
+/** Preview features that can be enabled on a mongo cluster. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+union PreviewFeature {
+  string,
+
+  /** Enables geo replicas preview feature. The feature must be set at create-time on new cluster to enable linking a geo-replica cluster to it. */
+  "GeoReplicas",
+}
+
+/** Parameters used for replica operations. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+model MongoClusterReplicaParameters {
+  /** The id of the replication source cluster. */
+  sourceResourceId: string;
```

Comment:
Can you use ArmResourceIdentifierAllowedResource or armResourceIdentifier instead of unrestricted string?